docs/design/versioning.md

```diff
@@ -34,19 +34,37 @@ Documentation for other releases can be found at
 
 # Kubernetes API and Release Versioning
 
+Reference: [Semantic Versioning](http://semver.org)
+
 Legend:
 
-* **Kube X.Y.Z** refers to the version of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.  (**X** is the major version, **Y** is the minor version, and **Z** is the patch version.)
+* **Kube X.Y.Z** refers to the version (tag) of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.  (**X** is the major version, **Y** is the minor version, and **Z** is the patch version.)
 * **API vX[betaY]** refers to the version of the HTTP API.
 
 ## Release versioning
 
 ### Minor version scheme and timeline
 
-* Kube X.Y.0-alpha.W, W > 0: Alpha releases are released roughly every two weeks directly from the master branch. No cherrypick releases. If there is a critical bugfix, a new release from master can be created ahead of schedule.
-* Kube X.Y.Z-beta.W: When master is feature-complete for Kube X.Y, we will cut the release-X.Y branch 2 weeks prior to the desired X.Y.0 date and cherrypick only PRs essential to X.Y.  This cut will be marked as X.Y.0-beta.0, and master will be revved to X.Y+1.0-alpha.0.  If we're not satisfied with X.Y.0-beta.0, we'll release other beta releases, (X.Y.0-beta.W | W > 0) as necessary.
-* Kube X.Y.0: Final release, cut from the release-X.Y branch cut two weeks prior.  X.Y.1-beta.0 will be tagged at the same commit on the same branch.  X.Y.0 occur 3 to 4 months after X.Y-1.0.
-* Kube X.Y.Z, Z > 0: [Patch releases](#patch-releases) are released as we cherrypick commits into the release-X.Y branch, (which is at X.Y.Z-beta.W,) as needed.  X.Y.Z is cut straight from the release-X.Y branch, and X.Y.Z+1-beta.0 is tagged on the same commit.
+* Kube X.Y.0-alpha.W, W > 0 (Branch: master)
+  * Alpha releases are released roughly every two weeks directly from the master branch.
+  * No cherrypick releases. If there is a critical bugfix, a new release from master can be created ahead of schedule.
+* Kube X.Y.Z-beta.W (Branch: release-X.Y)
+  * When master is feature-complete for Kube X.Y, we will cut the release-X.Y branch 2 weeks prior to the desired X.Y.0 date and cherrypick only PRs essential to X.Y.
+  * This cut will be marked as X.Y.0-beta.0, and master will be revved to X.Y+1.0-alpha.0.
+  * If we're not satisfied with X.Y.0-beta.0, we'll release other beta releases, (X.Y.0-beta.W | W > 0) as necessary.
+* Kube X.Y.0 (Branch: release-X.Y)
+  * Final release, cut from the release-X.Y branch cut two weeks prior.
+  * X.Y.1-beta.0 will be tagged at the same commit on the same branch.
+  * X.Y.0 occur 3 to 4 months after X.Y-1.0.
+* Kube X.Y.Z, Z > 0 (Branch: release-X.Y)
+  * Patch releases are released as we cherrypick commits into the release-X.Y branch, (which is at X.Y.Z-beta.W,) as needed.
+  * X.Y.Z is cut straight from the release-X.Y branch, and X.Y.Z+1-beta.0 is tagged on the same commit.
+* Kube X.Y.Z (Branch: release-X.Y.Z)
```

Comment:
Unrelated to this comment, I'm rethinking the -patch naming style.  Consider the following progression ( tag branch ):

Without using -patch:

1.1.0 release-1.1
1.1.1 release-1.1.0
1.1.2 release-1.1
1.1.3 release-1.1.0

Using -patch:

1.1.0              release-1.1
1.1.0-patch.1 release-1.1.0
1.1.1              release-1.1
1.1.2              release-1.1
1.1.0-patch.2 release-1.1.0

Using -patch provides information in the tag itself on which branch the tag lives.
Also the N+1 calculation can be made by simply looking at the last tag on any given branch where an activity is happening.  Without -patch we'd have to know the last N release across all branches to figure out what N+1 will be.

Now the tooling is not set up for branched patch releases yet, but we already have a note about this, so the handling of -patch in the naming can just be part of the TODO.

Now again, these kinds of releases are rare or non-existent depending on the paranoia level of really patching an existing release (vs. just continuously adding changes to release-X.Y), but we should distinguish here and incorporating the branch name into the tag is the right way to do this.

Another option is to allow for moving to X.Y.Z.W.  Either is fine and either will require changes to tooling - again, in the rare case we actually need to create an actual branched patch.